src/Text/Pandoc/Readers/LaTeX.hs

```diff
@@ -726,6 +728,14 @@ dolstinline = do
 doLHSverb :: PandocMonad m => LP m Inlines
 doLHSverb = codeWith ("",["haskell"],[]) <$> manyTill (satisfy (/='\n')) (char '|')
 
+dosiunitx :: PandocMonad m => LP m Inlines
+dosiunitx = do
+  skipopts
+  value <- char '{' >> manyTill anyChar (char '}')
```

Comment:
`braced` gives you a raw string inside the braces.  If you want LaTeX like `\euro` to be parsed, then use `tok`, which parses a LaTeX token (which might be a group like `{\euro 33}` but could also be e.g. a single digit).
This gives you a list of inlines, rather than a string.